File: app/src/main/java/br/com/jonathanzanella/myexpenses/source/SourceAdapterPresenter.java
package br.com.jonathanzanella.myexpenses.source;

import java.util.Collections;
import java.util.List;

class SourceAdapterPresenter {
	private final SourceRepository repository;

	private List<Source> sources;

	SourceAdapterPresenter(SourceRepository repository) {
		this.repository = repository;
		loadSources();
	}

	private void loadSources() {
		sources = repository.all();
	}

	List<Source> getSources(boolean invalidateCache) {
		if(invalidateCache)
			loadSources();
		return Collections.unmodifiableList(sources);
	}
}

File: app/src/main/kotlin/br/com/jonathanzanella/myexpenses/source/SourceAdapterPresenter.kt
package br.com.jonathanzanella.myexpenses.source

import java.util.*

class SourceAdapterPresenter(private val dataSource: SourceDataSource) {

    private var sources: List<Source>? = null

    init {
        loadSources()
    }

    private fun loadSources() {
        sources = dataSource.all()
    }

    fun getSources(invalidateCache: Boolean): List<Source> {
        if (invalidateCache)
            loadSources()
        return Collections.unmodifiableList(sources!!)
    }
}
